core/chains/types.go

```diff
@@ -0,0 +1,18 @@
+package chains
+
+// ID types represent unique identifiers within a particular chain type. Using string is recommended.
+type ID any
```

Comment:
IIUC it is not possible to define a constraint that would capture (`string` | `Stringer`), but we may be able to switch over to just using `string` later on, at least throughout the lower levels where we often stringify the `utils.Big`s to use as map keys anyways.

Reply:
I'm not sure if this would work here, but I think I've seen `~string` used in constraints to mean "kinda like a string" - not sure if that covers `Stringer` though.

Reply:
IIUC `~string` would allow us to use custom types with underlying type `string`, like if we just wanted to extend with additional methods. So it doesn't help with `utils.Big`, but we may want to use it eventually regardless.

Reply:
If we were OK limiting to uint64, then we could do a simpler version of `Ordered`, with just `~uint64 | ~string` or something :thinking: 
https://pkg.go.dev/golang.org/x/exp/constraints#Ordered